shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -145,6 +146,31 @@ static void DestroyJNI(JNIEnv* env, jobject jcaller, jlong shell_holder) {
   delete ANDROID_SHELL_HOLDER;
 }
 
+// Signature is similar to RunBundleAndSnapshotFromLibrary but it can't change
+// the bundle path or asset manager since we can only spawn with the same
+// AOT.
+static jlong SpawnJNI(JNIEnv* env,
+                      jobject jcaller,
+                      jobject flutterJNI,
```

Comment:
Rename this to spawnedJNI.  The flutterJNI name is a bit confusing because jcaller is also an instance of FlutterJNI.

Alternatively this method could use JNI APIs to construct the spawned FlutterJNI instance.

Reply:
I'm gonna call it newFlutterJNI. It's not technically "spawned" since nothing spawned it from an existing "pool". The Java side just constructed a new one and gave it to C++ for a callback. Adding some code comments too.

Reply:
> Alternatively this method could use JNI APIs to construct the spawned FlutterJNI instance.

I didn't understand this comment

Reply:
Currently `nativeSpawn` takes an argument containing an empty `FlutterJNI` representing the newly spawned engine.  `nativeSpawn` constructs the peer `AndroidShellHolder` and returns a pointer to it, which the Java side then stores in `nativeShellHolderId`.

The `nativeSpawn` C++ implementation could instead construct the new `FlutterJNI` instance, fill in its `nativeShellHolderId`, and return it.  That might make for a cleaner interface.  But it also means more complex JNI code.  So given that this is a private interface, it's probably more trouble than it's worth.

Reply:
You mean do something like https://stackoverflow.com/questions/10130819/use-jni-to-create-populate-and-return-a-java-class-instance right? Takes a bit of gymnastics but it is semantically clearer. Lower prio but it's interesting, I'll try it. The FlutterJNI construction signature is unlikely to change at this point too so we'll probably not expose Java refactors into entangled C++ changes.

Reply:
man that was really verbose, since the Java Long needs to be instantiated separately. But done now.